### DISCUSSION TITLE
Sunbird-Content-Editor - Setup

### DISCUSSION THREAD

**Discussion by kiran9ops** [Sunbird-Knowlg/Community#53] [Bugs]
Hi Team,

I tried setting up the Sunbird Content Editor following below link on my local Ubuntu machine, but I am not able to set it up properly.
https://github.com/project-sunbird/sunbird-content-editor

After final step of installation, enable 3000 port and able to access the link - http://localhost:3000/app

[http://localhost:3000/app?contentId={{content_id}}](http://localhost:3000/app?contentId=%7B%7Bcontent_id%7D%7D)

I am not able to figure out the content_id where I can get it from and the configurations file what all I need to update to access the link

Can you please help me out to resolve this configuration issue.

Thanks

**Comment by kiran9ops** [comment]
Can you please guide us to resolve the issue, thanks

**Comment by kiran9ops** [comment]
Can you please guide us to resolve the issue, thanks

**Comment by SumaGM2022** [comment]
@surendrasinghs Could you pls take a look at this? Kiran Bysani <kiran.bysani@trigyn.com> asking for an update.

**Comment by surendrasinghs** [comment]
@Sunbird-Knowlg/sunbirdknowlg_workinggroup please see.

@SumaGM2022 each building block has a working group handle. Please use that. Thanks!

**Comment by SumaGM2022** [comment]
Hi,
@Sunbird-Knowlg/sunbirdknowlg_workinggroup
<https://github.com/orgs/Sunbird-Knowlg/teams/sunbirdknowlg_workinggroup>This
takes to the page not found.
[image: image.png]

**Comment by kiran9ops** [comment]
Can you please guide us to resolve the issue, thanks

**Comment by vaibhavbhuva** [comment]
@dev9ops 

For more information, Please refer this documentation: https://knowlg.sunbird.org/learn/product-and-developer-guide/editors/editor

**Reply by kiran9ops** [reply]
We followed the installation procedure given in the link provided but not aware of the values what we need provide in the configuration

Required configuration
------------------------------	
window.context = {
{
  "user": {
    "id": " ",
    "name": "Admin",
    "orgIds": [  ],
    "organisations": {}
  },
  "sid": "xxx",
  "contentId": "do_xxx",
  "pdata": {
    "id": "",
    "ver": "1.11.0",
    "pid": ""
  },
  "tags": [],
  "channel": "xxx",
  "framework": "NCFCOPY",
  "ownershipType": [
    "createdBy",
    "createdFor"
  ],
  "uid": "xxx",
  "etags": {
    "app": [],
    "partner": [],
    "dims": []
  }
};
window.config = {
  "baseURL": "",
  "apislug": "/action",
  "build_number": "1.11.0.1bb2ae8",
  "pluginRepo": "/content-plugins",
  "aws_s3_urls": [],
  "plugins": [
    {
      "id": "org.ekstep.sunbirdcommonheader",
      "ver": "1.6",
      "type": "plugin"
    }
  ],
  "corePluginsPackaged": true,
  "dispatcher": "local",
  "localDispatcherEndpoint": "",
  "previewURL": "/content/preview/preview.html",
  "modalId": "contentEditor",
  "alertOnUnload": true,
  "headerLogo": "https://dev.open-sunbird.org/assets/images/sunbird_logo.png",
  "showHelp": false,
  "previewConfig": {},
  "pluginsRepoUrl": " ",
  "enableTelemetryValidation": false
}
--------------------------------------------------------
  openContentEditor() {
    jQuery.fn.iziModal = iziModal;
    jQuery('#content-editor').iziModal({
      title: '',
      iframe: true,
      iframeURL: 'url', // content-editor node_moduels index.html path
      navigateArrows: false,
      fullscreen: false,
      openFullscreen: true,
      closeOnEscape: false,
      overlayClose: false,
      overlay: false,
      overlayColor: '',
      history: false,
      onClosing: () => {
        this._zone.run(() => {
          this.closeModal();
        });
      }
    });

**Reply by kiran9ops** [reply]
Can you please guide us to resolve the issue, thanks

**Reply by vinukumar-vs** [reply]
@dev9ops What configuration options are you not able to define/update? Can you please share the specific details what configurations are not able to understand?

The details of each configuration property have detailed in the configuration table.
https://github.com/project-sunbird/sunbird-content-editor/blob/release-5.0.0/README.md#how-to-configure-the-sunbird-content-editor

If you are looking for "content_id" in the below URL. You should have your dev sunbird evnrionment setup ready & you are taking the contentID which is part of your dev environment(The content has to created which empty data using API, You are referring that content(content_id in the URL) to edit using the content-editor)
[http://localhost:3000/app?contentId={{content_id}}](http://localhost:3000/app?contentId=%7B%7Bcontent_id%7D%7D)

You can use the below configuration to do that
**sample:**
{
        baseURL: 'https://dev.{hostName}}/',
	apislug: '/{{API proxy slug}}',
} 

@gourav.more Please add if I am missing any other information which is required for local setup to run.

**Comment by gouravmore** [comment]
@dev9ops Is the above comment solved your local installation issue?

**Reply by kiran9ops** [reply]
Hi Gourav, 
Completed all below 8 steps. 
How to Setup Sunbird Content Editor in your Local Machine
1.Clone the sunbird-content-editor repo from [here](https://github.com/project-sunbird/sunbird-content-editor)
2.Clone the sunbird-content-plugins repo from [here](https://github.com/project-sunbird/sunbird-content-plugins)
3.Go to the root directory sunbird-content-editor.
4.Run npm install to install node modules. 
5.cd app and run bower install to install bower components
6.Create a symlink to 'sunbird-content-plugins' (ln -s ../sunbird-content-plugins plugins)(Linux, mac) for windows: use mklink
7.Run node app
8.Open Chrome and visit this link: [http://localhost:3000/app?contentId={{content_id}}](http://localhost:3000/app?contentId=%7B%7Bcontent_id%7D%7D)
--------------------------------------------------------------------------------------
Want to know what FILES and location of those files to update the configuration to access the link.

Can we have a call so that I could explain in better way.

Thanks,

**Reply by kiran9ops** [reply]
Hi Team, Can you please guide us to resolve the issue, thanks

**Reply by gouravmore** [reply]
Sure @dev9ops we can have a call.You can update the config in following file path.
File Location :- /sunbird-content-editor/app/scripts/contenteditor/ce-config.js

**Comment by kiran9ops** [comment]
[ce-config.js.txt](https://github.com/Sunbird-Knowlg/Community/files/9613356/ce-config.js.txt)

I have uploaded request file, please provide support contact number.

**Reply by gouravmore** [reply]
As per above ce-config.js file - content editor should work. Can you please share the following details.
Content id:
Env name: 
Branch name of sunbird-content-plugins and sunbird-content-editor:
ScreenShot of console if any error

**Reply by kiran9ops** [reply]
![Login Host](https://user-images.githubusercontent.com/96770317/191522202-2f4dfae9-aeaa-4a6e-b853-4bd140cbb303.JPG)

its Ubuntu environment 
find the screenshot, page shows loading when trying to access the site.

**Reply by gouravmore** [reply]
@dev9ops Is there any error in console? can you please attach the screen shot of console also?
Can you please share the following details with the screenshots: 
What is the content id you are using?
In which environment you have created the content?
What is the Branch name of sunbird-content-plugins and sunbird-content-editor?

**Reply by kiran9ops** [reply]
Gourav, Can we have call please

**Reply by kiran9ops** [reply]
1) Content ID's
        "identifier": "do_2****************************1", [Environmental Studies]
        "objectType": "Content"
        "identifier": "do_2******************************2", [Math]
        "objectType": "Content"

2) Education

3) Followed below steps
        How to Setup Sunbird Content Editor in your Local Machine
        1.Clone the sunbird-content-editor repo from here
        2.Clone the sunbird-content-plugins repo from here
        3.Go to the root directory sunbird-content-editor.
        4.Run npm install to install node modules.
        5.cd app and run bower install to install bower components
        6.Create a symlink to 'sunbird-content-plugins' (ln -s ../sunbird-content-plugins plugins)(Linux, mac) for windows: use mklink
        7.Run node app

4)  I am able to access http://localhost:3000/app but page not fully loaded

5)  Which configuration files i have to update to access the link fully

6) What is the Branch name of sunbird-content-plugins and sunbird-content-editor?
   Can you brief me what branches you are referring to.

**Can we have a call** to fix this... its been long time issue pending

**Reply by gouravmore** [reply]
@dev9ops  
To load the content - you should add the content id in query parameter. Please use below URL to load the content editor locally.
URL:- _http://localhost:3000app/?contentId=do_21_

**Reply by kiran9ops** [reply]
Hi... i have to add the content ID in some configuration file where it will be accessible, please let me know the path and the file name.

**Reply by gouravmore** [reply]
@dev9ops No need to add the content ID in any configuration. The content editor will automatically take it from query parameter. Please use the bellow URL to access the content editor locally.
URL:- http://localhost:3000app/?contentId=do_21

**Reply by kiran9ops** [reply]
http://localhost:3000/app/?contentId=do_21*********************

page partially loading

**Reply by gouravmore** [reply]
Can you please add the screenshot of it, with the developer console opened.

**Reply by kiran9ops** [reply]
![image](https://user-images.githubusercontent.com/96770317/196180027-c8ef6dda-2589-45aa-8691-68bfd28ad71d.png)

**Reply by kiran9ops** [reply]
screenshot attached

**Reply by gouravmore** [reply]
Still i'm not able to see developer console in screenshot. If you are using chrome browser - press => Shift + ctrl + i and share the screenshot of it

**Reply by kiran9ops** [reply]
![Sunbird error 1](https://user-images.githubusercontent.com/96770317/196363464-627db7d6-52d9-4e3a-bb9c-afd22edbe4df.JPG)

**Reply by kiran9ops** [reply]
screenshot attached

**Reply by kiran9ops** [reply]
can you please look into the screenshot and help to fix

**Reply by kiran9ops** [reply]
Team, can you please check the screenshot and guide to resove

**Reply by gouravmore** [reply]
@dev9ops 
Please share the following details
- Content URL (which you are trying to access it locally)
- How you are getting this content id (from which environment )

**Reply by kiran9ops** [reply]
We are following below links to get content ID

http://docs.sunbird.org/latest/apis/content/index.html#operation/Search%20Content
https://api.dev.diksha.gov.in/#/discoverapi

**Reply by gouravmore** [reply]
@dev9ops In which environment you are using this API. Please share the Content URL (which you are trying to access it locally). Without content URL - we cannot debug this issue. This might be a content specific issue. Is this content loading correctly in their respective environment?

**Reply by kiran9ops** [reply]
![image](https://user-images.githubusercontent.com/96770317/199163412-283a8861-ad19-482c-a651-c8406f846b2d.png)

**Reply by kiran9ops** [reply]
Can you schedule meeting or provide contact number to sort it out

**Reply by swapnila79** [reply]
> @dev9ops In which environment you are using this API. Please share the Content URL (which you are trying to access it locally). Without content URL - we cannot debug this issue. This might be a content specific issue. Is this content loading correctly in their respective environment?
Hello Team,
we are trying to use the sunbird editor for our web application, So are there any open content ID available which we can use to display the editor in browser.

**Reply by kiran9ops** [reply]
Team @gouravmore can you please update on @swapnila79 request

**Reply by gouravmore** [reply]
Please share your email id or provide a contact number to connect.

**Comment by swapnila79** [comment]
Hello Gourav,

Please find my email id as
***@***.******@***.***>

Thanks,
Swapnil

**Reply by kiran9ops** [reply]
@gouravmore send meeting invite to devopsroom@gmail.com

**Reply by kiran9ops** [reply]
@gouravmore As discussed last day could you please update support status from Admin team

**Reply by gouravmore** [reply]
@kiran9ops As per our last day call, You don't have a proper back-end setup to get the content ids. To get the content Id, you have the following options
- For local setup, you can use the sunbird instance content ids. For these ids - you have to request for a token to get the existing Sunbird instance ids
- Do the entire backend setup and generate the token and create the content in your own server setup
cc- @vinukumar-vs @vaibhavbhuva

**Reply by kiran9ops** [reply]
@vinukumar-vs @vaibhavbhuva @gouravmore
Hi Team, please initiate a meeting devopsroom@gmail.com we can setup the environment as discussed. Thanks

**Reply by vaibhavbhuva** [reply]
@kiran9ops Please refer to this documentation for sunbird installation: https://ed.sunbird.org/use/prerequisites-for-your-own-sunbird-ed-instance.
For your information, there is a sunbird knowledge demo application available online where you can see and play it with all the editors and players as well. Here is the demo application link: https://dev.knowlg.sunbird.org/. Please go through it.

Please reach out to https://github.com/orgs/Sunbird-Ed/discussions/ in case of any queries. Thanks

cc: @vinukumar-vs  @keshavprasadms

**Reply by kiran9ops** [reply]
@vinukumar-vs @vaibhavbhuva @gouravmore
Hi Team,
Can you please provide basic configuration installation setup, as below link needs high-end configuration
https://ed.sunbird.org/use/prerequisites-for-your-own-sunbird-ed-instance.

**Reply by kiran9ops** [reply]
@vinukumar-vs @vaibhavbhuva @gouravmore
Hi Team,
Can you please provide basic configuration installation setup, as below link needs high-end configuration
https://ed.sunbird.org/use/prerequisites-for-your-own-sunbird-ed-instance.
